packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -73,74 +82,99 @@ export class BlockToolAdapter {
    * @param input - input element
    * @param key - data key input is attached to
    */
-  #handleBeforeInputEvent(event: InputEvent, input: HTMLElement, key: DataKey): void {
+  #handleBeforeInputEvent = (event: InputEvent, input: HTMLElement, key: DataKey): void => {
```

Comment:
If you want to save context here, would be better to use an anonymous arrow function when adding a listener. Arrow function methods are treated as properties, therefore created for each instance (which we expect a lot for a document)